content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,501 @@
+---
+title: Troubleshooting Application Security Monitoring
+kind: documentation
+further_reading:
+- link: "/security_platform/application_security/"
+  tag: "Documentation"
+  text: "Monitoring Threats with Datadog Application Security"
+- link: "/security_platform/application_security/getting_started/"
+  tag: "Documentation"
+  text: "Get Started Using Application Security to Detect Threats"
+- link: "/security_platform/application_security/setup_and_configure/#compatibility"
+  tag: "Documentation"
+  text: "Programming Language and Framework Compatibility"
+---
+
+If you experience unexpected behavior with Datadog Application Security, there are common issues you can investigate, as mentioned below. If you continue to have trouble, reach out to [Datadog support][1] for further assistance. 
+
+## Application Security rate limits
+
+Application Security traces are rate-limited to 100 traces per second. Traces sent after the limit are not reported. Contact [Datadog support][1] if you need to change the limit.
+
+## No Suspicious Requests detected by Application Security
+
+There are a series of steps that must run successfully for threat information to appear in the Application Security [Trace and Signals Explorer][2]. It is important to check each step when investigating this issue. Additional troubleshooting steps for specific languages are in the language tab at the end.
+
+### Confirm Application Security is enabled
+
+You can use the metric `datadog.apm.appsec_host` to check if Application Security is running.
+
+1. Go to **Metrics > Summary** in Datadog.
+2. Search for the metric `datadog.apm.appsec_host`. If the metric doesn’t exist, then there are no services running Application Security. If the metric exists, the services are reported with the metric tags `host` and `service`.
+3. Select the metric, and in the **Tags** section, search for `service` to see which services are running Application Security.
+
+If you are not seeing `datadog.apm.appsec_host`, check the [in-app instructions][3] to confirm that all steps for the initial setup are complete.
+
+Application Security data is sent with APM traces. See [APM troubleshooting][4] to [confirm APM setup][5] and check for [connection errors][6].
+
+### Send a test attack to your application
+
+ To test your Application Security setup, trigger the [Security Scanner Detected][7] rule by running a file that contains the following curl script: 
+
+{{< programming-lang-wrapper langs="java,.NET,go,ruby,PHP,NodeJS" >}}
+{{< programming-lang lang="java" >}}
+
+```bash
```

Comment:
Should most of these code examples be the same across all languages? Only difference I'm noticing is `Arachni/v1.0` vs. `dd-test-scanner-log`.

Reply:
Note: I usually use `Nessus SOAP` for Ruby ;)

The key difference is:

- `Arachni/v1.0` is a real-life malicious user agent but might be caught by an earlier security system.
- `dd-test-scanner-log` is a canary user agent specific to Datadog so should not be caught by an earlier security system and is much more guaranteed to work, but feels "less real" to customers. Also `dd-test-scanner-log` may not be in all library variants yet (to be confirmed).

I seem to recall we want both examples ("real life" one + "should always work" one as fallback) and that they should be unified across languages, but this should probably be validated with each library owner.